File: js/sumPairs.js
exports.sumPairs = function(arr, num) {
  let result = [];
  let total = 0;
  // Check for input type
  if (!(arr instanceof Array)) {
    return "Not a list"
  } else if (!(typeof num === 'number')) {
    return "Not a number"
  }
  let options = {}
  for (let i=0; i<arr.length; i++) {
    if ((num - arr[i]) in options) {
      result.push([num-arr[i], arr[i]])
    }
    arr[i] in options ? options[arr[i]] += 1 : options[arr[i]] = 1
    
  }
  if (result.length == 0) {
    return "Unable to find pairs"
  }
  return result
};

File: python/sum_pairs.py
def sum_pairs(numbers, total):
    result = []
    # Check for wrong input type
    if not isinstance(numbers, list):
        return "Not a list"
    elif not (isinstance(total, int) or isinstance(total, float)):
        return "Not a number"
    numsdict = {}
    for num in numbers:
        if (total - num) in numsdict:
          result.append([total - num, num])
        numsdict[num] = numsdict[num] + 1 if num in numsdict else 1
    if result == []:
        return "Unable to find pairs"
    return result